tests/test_gaap.py

```diff
@@ -167,21 +190,92 @@ def check(self, psfSigma=0.5, flux=1000., scalingFactors=[1.15], forced=False):
 
-    def testGaapForced(self):
-        self.runGaap(True)
+        This is an alternative implementation to the `_getFluxErrScaling`
+        method of `BaseGaapFluxPlugin`, but is less efficient.
+
+        Parameters
+        ----------
+        `kernel` : `lsst.afw.math.Kernel`
+            The PSF-Gaussianization kernel.
+
+        Returns
+        -------
+        fluxErrScaling : `float`
+            The factor by which the standard error on GAaP flux must be scaled.
+        """
+        kim = afwImage.ImageD(kernel.getDimensions())
+        kernel.computeImage(kim, False)
+        weight = galsim.Image(np.zeros_like(kim.array))
+        aperSigma = aperShape.getDeterminantRadius()
+        gauss = galsim.Gaussian(sigma=aperSigma, flux=2*np.pi*aperSigma**2)
+        weight = gauss.drawImage(image=weight, scale=1.0, method='no_pixel')
+        kwarr = scipy.signal.convolve2d(weight.array, kim.array, boundary='fill')
+        fluxErrScaling = np.sqrt(np.sum(kwarr*kwarr))
+        fluxErrScaling /= np.sqrt(np.pi*aperSigma**2)
+        return fluxErrScaling
+
+    def testCorrelatedNoiseError(self, sigmas=[3.0, 4.0], scalingFactors=[1.15, 1.2, 1.25, 1.3, 1.4]):
+        """Test the scaling to standard error due to correlated noise.
+
+        The uncertainty estimate on GAaP fluxes is scaled by an amount
+        determined by the auto-correlation function of the PSF-matching kernel;
+        see Eqs. A11 & A17 of Kuijken et al. (2015). This test ensures that the
+        calculation of the scaling factors matches the analytical expression
+        when the PSF-matching kernel is a Gaussian.
+        """
+        # Create an image of an extended source
+        gaapConfig = lsst.meas.extensions.gaap.GaapFluxConfig(sigmas=sigmas, scalingFactors=scalingFactors)
+        gaapConfig.scaleByFwhm = True
+
+        algorithm, schema = self.makeAlgorithm(gaapConfig)
+        exposure, catalog = self.dataset.realize(0.0, schema, randomSeed=123456789)
+        record = catalog[0]
+        center = self.center
+        seeing = exposure.getPsf().computeShape(center).getDeterminantRadius()
+        for sF in gaapConfig.scalingFactors:
+            targetSigma = sF*seeing
+            modelPsf = afwDetection.GaussianPsf(algorithm.config.modelPsfDimension,
+                                                algorithm.config.modelPsfDimension,
+                                                targetSigma)
+            result = algorithm._generic._convolve(exposure, modelPsf, record)
+            kernel = result.psfMatchingKernel
+            kernelAcf = algorithm._generic._computeKernelAcf(kernel)
+            for sigma in gaapConfig.sigmas:
+                aperSigma2 = sigma**2 - targetSigma**2
+                aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0)
+                fluxErrScaling1 = algorithm._generic._getFluxErrScaling(kernelAcf, aperShape)
+                fluxErrScaling2 = self.getFluxErrScaling(kernel, aperShape)
```

Comment:
This gets an alternate implementation of the `fluxErrScaling` number, but using the RHS of Eq. 12 in the tech note.